src/shredder.rs

```diff
@@ -255,24 +255,25 @@ impl Shredder for RegularShredder {
 
     fn shred(slice: Slice, sk: &SecretKey) -> Result<[ValidatedShred; TOTAL_SHREDS], ShredError> {
         let (header, payload) = slice.deconstruct();
-        let raw_shreds = reed_solomon_shred(payload.into(), Self::CODING_OUTPUT_SHREDS)?;
+        let raw_shreds = reed_solomon_shred(&payload.to_bytes(), Self::CODING_OUTPUT_SHREDS)?;
```

Comment:
we don't save much here as we are having to convert to a vec here but it seems like we have savings in when we call `reed_solomon_shred` in the `deshred` function below where we do not need to convert to a vec now.